phone-book/src/app.js

```diff
@@ -0,0 +1,177 @@
+    this.appContainer.innerHTML = appHtml;
+    this.addRouter();
+
+    this.state.pageName = 'Contacts';
+    this.updateState({activePage : 'contacts'});
+    this.router.gotToPage(this.state,'/index.html');
+    this.changePageToActive();
+  }
+
+  updateState(newState){
+    this.state = {};
+    Object.assign(this.state,newState);
+
+
+  }
+
+  addRouter(){
+    this.appContainer.addEventListener('click', (event) => {
+      event.preventDefault();
+      const container = event.currentTarget;
+      let target = event.target;
+      while (target != container) {
+        if (target.getAttribute("data-page") ) {
+          const newPage = target.getAttribute("data-page");
+          const newUser = target.getAttribute("data-user-id");
+          const newState = { activePage : newPage };
+          if(newUser){
+            newState.userId = newUser;
+          }
+          this.updateState(newState);          
+          this.router.gotToPage(this.state);
+          this.changePageToActive();
+          return;
+        }
+        target = target.parentNode;
+      }
+    });  
+  }
+
+  changePageToActive(){
+    this.updateTitle();
+    this.insertCurrentPage();
+    this.initPageEventListeners();
+  }
+  updateTitle(){    
+    const currentPage = this.getCurrentPage(); 
+    const title = this.appContainer.querySelector('header > div > h2');
+    title.textContent = currentPage.title;
+  }
+
+  getCurrentPage(){
+    const activePage = this.state.activePage;
+    const currentPage = this.pages[activePage].pageObject;  
+    return currentPage;  
+  }
+
+  initPageEventListeners(){
+    const currentPage = this.getCurrentPage(); 
+    currentPage.initEvents();
+  }
+  
+  renderAppContainer(){
+    return `
+    ${this.renderAppHead()}
+    ${this.renderAppPage()}
+    ${this.renderAppFooter()}    
+    `;
+  }
+
+  formatformNumber(number) {
+    if(isNaN(number.charAt(0))){
+      return number;
+    }
+    const formatedNumber = number.replace(/(\d{0,3})(\d{0,2})?(\d{0,2})?(\d{0,3})?/, (match,g1,g2,g3,g4) => {
+      let res = ''
+      if(g1){
+        res = `(${g1}`;
+      }
+      if(g1.length === 3){
+        res += `) `;
+      }
+      if(g2){
+        res += `${g2}`;
+      }
+      if(g3){
+        res += `-${g3}`;
+      }
+      if(g4){
+        res += `-${g4}`;
+      }
+      return res
+    });    
+    return formatedNumber;
+  }
+
+  renderAppHead(){
+    return `
+        <header class="header">
+            <div class="container top-radius">
+                <h2></h2>
+            </div>
+        </header>   
+    `;
+  }
+
+  renderAppFooter(){
+    return `
+    <footer class="footer">
+        <div class="container bottom-radius">
+            <nav class="main-nav">
+                ${this._createFooterIcon( { href: "index.html", page:"contacts", title: "Contacts",icon: "search", active: true})}
+                ${this._createFooterIcon( { href: "keypad.html", page:"keypad", title: "Keypad",icon: "th",})}
```

Comment:
please use links from this.pages..href 
It's hard to understand why you contacts here has an a href equal to `index.html` but inside this.pages(probably it's source of truth) it refers to `contacts.html`